kinto_http/tests/test_functional_async.py

```diff
@@ -0,0 +1,621 @@
+import hashlib
+import hmac
+from typing import Dict, Tuple
+from urllib.parse import urljoin
+
+import pytest
+import requests
+from pytest_mock import MockerFixture
+
+from kinto_http import AsyncClient as Client
+from kinto_http import BucketNotFound, CollectionNotFound, KintoException
+from kinto_http.patch_type import JSONPatch
+
+
+SERVER_URL = "http://localhost:8888/v1"
+DEFAULT_AUTH = ("user", "p4ssw0rd")
+
+
+# Backported from kinto.core.utils
+def hmac_digest(secret, message, encoding="utf-8") -> hmac.HMAC:
+    """Return hex digest of a message HMAC using secret"""
+    if isinstance(secret, str):
+        secret = secret.encode(encoding)
+    return hmac.new(secret, message.encode(encoding), hashlib.sha256).hexdigest()
+
+
+def create_user(server_url: str, credentials: Tuple[str, str]) -> Dict:
+    account_url = urljoin(server_url, "/accounts/{}".format(credentials[0]))
+    r = requests.put(account_url, json={"data": {"password": credentials[1]}}, auth=DEFAULT_AUTH)
+    r.raise_for_status()
+    return r.json()
+
+
+def get_user_id(server_url: str, credentials: Tuple[str, str]) -> str:
+    r = create_user(server_url, credentials)
+    return f"account:{r['data']['id']}"
+
+
+@pytest.mark.asyncio
+async def test_bucket_creation(functional_setup):
+    client = functional_setup
+    bucket = await client.create_bucket(id="mozilla")
+    user_id = get_user_id(SERVER_URL, DEFAULT_AUTH)
```

Comment:
maybe we could do something like `get_user_id(client.server_url, client.session.auth)`?